New implementation tests: address pylint warnings

MVrachev opened the pull request:
Related to #1657:

**Description of the changes being introduced by the pull request**:

I believe we should enable linting (tox -e lint) for new test files testing the new implementation.

The first step to do that would be to apply the automatic fixes proposed
by `black` and `isort` on the new test files testing the new code.
This happened in #1658.

The next step would be to propose fixes for `pylint` and `mypy` and the last
the part will be to enable the linters themselves.

In this pr, I include fixes for the warnings by `pylint` only as those changes are **NOT**
automatically by the tool and are more than enough for a pr.

Commit:
Pylint: disable duplicate-code for the new code

Pylint reported a couple of warnings flagged as "duplicate-code".
We were truly duplicating code - one of the examples was when we
imported the same objects from tuf/api/metadata.py:
MetaFile, Role, Root, Snapshot, TargetFile, Targets, and Timestamp
in two separate modules.
So, I thought we do want to be repetitive here and include that code at
both modules. The problem is that besides importing the above
classes the modules imported other classes from tuf.api.metadata.py
and there was no way to disable this check.
I searched and found out that this is a known problem:
https://github.com/PyCQA/pylint/issues/214.

That's why the only solution I see is to disable this warning
temporarily and hoping that one day when this issue is fixed we will
remember to turn it on again.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

Commit:
Address pylint warnings on tests for the new code

Address or disable pylint warnings raised on all test files inside
the "tests/" directory testing the code of the new implementation.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

Commit:
Pylint config: add _ as a good variable name

_ is often used when a function returns multiple values and you need
a sub-portion of them. Then, those values that are unnecessary can be
named _.
Currently, pylint warns us that this is not a good variable name, so
fix that.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

kairoaraujo commented on tests/repository_simulator.py:
```diff
@@ -136,6 +139,7 @@ def targets(self) -> Targets:
         return self.md_targets.signed
 
     def all_targets(self) -> Iterator[Tuple[str, Targets]]:
+        """Yield role name and signed portion of targets one by one"""
```
I am not sure if relevant comment.  
To be consistent with another docstring you had added, finish with a period.
If it makes sense, it would be interesting to apply the other changes that are part of this PR.

MVrachev replied:
I will add `.` at the end of the docstrings.
Do you have a better suggestion for this docstring? 
The function is simple and I am not sure what else can I write here.

kairoaraujo replied:
I don't have a better suggestion for this docstring. LGTM

[General comment, not sure if the right place]
I think what we are doing in the ``ngclient`` for docstring is the best option. In a sense, it can become good documentation in the end (as part of readthedocs/sphinx).
I understand that this is a test part, but it can help a lot of new contributors. Any thoughts @jku ?

kairoaraujo commented on tests/repository_simulator.py:
```diff
@@ -185,6 +189,9 @@ def publish_root(self):
         logger.debug("Published root v%d", self.root.version)
 
     def fetch(self, url: str) -> Iterator[bytes]:
+        """Fetches data from the given url and
```
See below for a suggestion to add more content within 80 columns line. (optional, I guess)

```python
        """Fetches data from the given url and returns an Iterator (or yields
        bytes).
        """
```

MVrachev replied:
I tried splitting it where it made sense logically, but that's a personal taste.
Will do it as you suggested.

kairoaraujo commented on tests/repository_simulator.py:
```diff
@@ -234,35 +246,35 @@ def _fetch_metadata(
             # return a version previously serialized in publish_root()
             if version is None or version > len(self.signed_roots):
                 raise FetcherHTTPError(f"Unknown root version {version}", 404)
```
Shall we log something before raising the ``FetcherHTTPError``? 
If it makes sense would be good to apply also to lines 263, 265

jku replied:
I don't think this is needed here: the repository (or the FetcherInterface implementation to be exact) is saying it does not have this metadata, that's it (we also have zero other details to log)

kairoaraujo commented on tests/repository_simulator.py:
```diff
@@ -274,6 +286,7 @@ def _compute_hashes_and_length(
         return hashes, len(data)
 
     def update_timestamp(self):
+        """Update timestamp and assign snapshot ver to snapshot_meta ver"""
```
Add the snapshot_meta between simple quotes. I would suggest being explicit and using version instead ver.

MVrachev replied:
I used "ver" instead of "version" in order to keep it a one-line docstring as this is a short function, but if it's unclear will change it.

kairoaraujo replied:
It is clear, but IMHO, I would pick up the exact reference in the documentation between a one-line docstring and an explicit reference. 
I still see my comment as optional as it is test documentation.

kairoaraujo commented on tests/test_api.py:
```diff
@@ -44,6 +44,8 @@
 
 
 class TestMetadata(unittest.TestCase):
+    """Tests for public API of all classes in tuf/api/metadata.py"""
```
Add the tuf/api/metadata.py in between single quotes.

kairoaraujo commented on tests/test_api.py:
```diff
@@ -155,8 +157,8 @@ def test_to_from_bytes(self):
             self.assertEqual(metadata_obj_2.to_bytes(), obj_bytes)
 
     def test_sign_verify(self):
-        root_path = os.path.join(self.repo_dir, "metadata", "root.json")
-        root = Metadata[Root].from_file(root_path).signed
+        path = os.path.join(self.repo_dir, "metadata", "root.json")
+        root = Metadata[Root].from_file(path).signed
```
Above you changed from ``dir`` to ``dest_dir``, I see it as more explicit.
At these lines, I see the ``root_path`` as ``path``, any specific reason for this?

jku replied:
dir was likely changed because it's a builtin, this change is likely to keep lines below 80 chars

MVrachev replied:
`root_path` was changed to `path`, because there was a pylint warning we were using too many variables in a single function and with this change, we can reuse `path`.
About `dir` Jussi is correct.

jku replied:
re-using variables is usually a code smell but this looks OK to me. The best solution might be a refactor of the function (to maybe multiple tests) but I appreciate not doing large changes under "address pylint warnings": anyway LGTM

MVrachev replied:
This is no longer of a concern after the merge of #1671 and I removed this change.

kairoaraujo commented on tests/test_updater_ng.py:
```diff
@@ -17,20 +17,22 @@
 
 
 class TestUpdater(unittest_toolbox.Modified_TestCase):
+    """Test the Updater class from tuf.ngclient.updater.py"""
```
I'm not sure that the '.py' is needed. Add between simple quotes (?).

MVrachev replied:
Yea, well it's doesn't really matter, I can change it, but I am not sure if there is some benefit in doing so.
I think I have used it in other places as well.

kairoaraujo replied:
Indeed. My comment was more about having a single standard for consistency.

MVrachev commented:
I addressed comments made by @kairoaraujo:
- added `.` at the end of the docstrings
- used one consistent format when mentioning a specific python file
- small docstring improvements

jku reviewed (approved):
Thanks Martin, Kairo. LGTM.

jku commented:
I was a bout to merge ... but first merged another change that created a conflict. Sorry about that, could you resolve please?

MVrachev commented:
Rebased on top of develop as there were conflicts after the merge of #1636.

MVrachev commented:
I proposed a pr fixing pylint warning about too many local variables in `test_sign_verify()` by splitting it: #1671.
Maybe we can merge #1671 and then I will update this one without reusing `path`?

MVrachev commented:
#1671 is merged and I have rebased and removed the change which handled too many local variables (renaming `root_path` to `path`).
